services/spar/test/Test/Spar/Sem/ScimExternalIdStoreSpec.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module Test.Spar.Sem.ScimExternalIdStoreSpec where
+
+import Arbitrary ()
+import Imports
+import Polysemy
+import Polysemy.Check
+import qualified Spar.Sem.ScimExternalIdStore as E
+import Spar.Sem.ScimExternalIdStore.Mem
+import Test.Hspec
+import Test.Hspec.QuickCheck
+import Test.QuickCheck
+
+deriveGenericK ''E.ScimExternalIdStore
+
+propsForInterpreter ::
+  PropConstraints r f =>
+  String ->
+  (forall a. Sem r a -> IO (f a)) ->
+  Spec
+propsForInterpreter interpreter lower = do
```

Comment:
why is it called "lower"?

Reply:
because it lowers the effect into IO and `run` was already taken :)

Reply:
ah, that makes sense.  :)  *much* better than `run'`!